lib/travis/api/app/endpoint/authorization.rb

```diff
@@ -300,7 +300,7 @@ def user_for_github_token(token, drop_token = false)
         end
 
         def get_token(endpoint, values)
-          response   = Faraday.new(ssl: Travis.config.github.ssl || {}).post(endpoint, values)
+          response   = Faraday.new(ssl: Travis.config.ssl.to_h.merge(Travis.config.github.ssl || {}).to_h.compact).post(endpoint, values)
```

Comment:
Seems like this second `.to_h` is probably not needed since merge returns a `Hash`.